ukis_pysat/data.py

```diff
@@ -248,15 +305,17 @@ def download_image(self, platform, product_uuid, target_dir):
             # query EarthExplorer for srcid of product
             meta_src = self.api.request("metadata", **{"datasetName": platform.value, "entityIds": [product_uuid],},)
             product_srcid = meta_src[0]["displayId"]
-            # download data from AWS
-            product = Product(product_srcid)
-            product.download(out_dir=target_dir, progressbar=False)
 
-            # compress download directory and remove original files
-            shutil.make_archive(
-                target_dir.joinpath(product_srcid), "zip", root_dir=target_dir.joinpath(product_srcid),
-            )
-            shutil.rmtree(target_dir.joinpath(product_srcid))
+            if not Path(target_dir.joinpath(product_srcid + ".zip")).exists():

```

Comment:
Wouldn't [`is_file()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file) be more what you want to do here?

Reply:
Great, yes thanks that's much better.